index.php

```diff
@@ -198,10 +198,25 @@ private function render_footer() {
     print('<p class="text-center">');
 
     if ($this->frontpage['show_visitor_ip']) {
-      if (isset($_SERVER['HTTP_X_FORWARDED_FOR'])) {
-        print('Your IP address: '.htmlentities($_SERVER['HTTP_X_FORWARDED_FOR']).'<br>');
+      if (
+        $this->misc['enable_http_x_forwarded_for'] === true
+        &&
+        isset($_SERVER['HTTP_X_FORWARDED_FOR'])
+      ) {
+        // The user can pass several proxy's, which each one will add its own IP address,
+        //  so we like to take only the first IP address
+        $ips = explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']);
+        $ip = trim($ips[0]);
+        $requester = filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_IPV4);
+        if ($requester === false) {
+          // fall back to REMOTE_ADDR
+          $ip = $_SERVER['REMOTE_ADDR'];
+        } else {
+          $ip = $_SERVER['REMOTE_ADDR'];
+        }
```

Comment:
The statement seems a bit odd. It does the same thing whatever will be evaluated.

Reply:
You are right, this was a bad C'n'P from the `execute.php` file.